source/docs/software/commandbased/convenience-features.rst

```diff
@@ -1,290 +1,343 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-In order to inline a command definition, users require some way to
-specify what code the commands will run as constructor parameters.
-Fortunately, both Java and C++ offer users the ability to pass
-subroutines as parameters.
+In order to inline a command definition, users require some way to specify what code the commands will run as constructor parameters. Fortunately, both Java and C++ offer users the ability to pass subroutines as parameters.
 
 Method references (Java)
-^^^^^^^^^^^^^^^^^^^^^^^^
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-In Java, a reference to a subroutine that can be passed as a parameter
-is called a method reference. The general syntax for a method reference
-is ``object::method``. Note that no method parameters are included,
-since the method *itself* is the parameter. The method is not being
-called - it is being passed to another piece of code (in this case, a
-command) so that *that* code can call it when needed. For further
-information on method references, see `the official Oracle
-documentation <https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html>`__.
+In Java, a reference to a subroutine that can be passed as a parameter is called a method reference. The general syntax for a method reference is ``object::method``. Note that no method parameters are included, since the method *itself* is the parameter. The method is not being called - it is being passed to another piece of code (in this case, a command) so that *that* code can call it when needed. For further information on method references, see `the official Oracle documentation <https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html>`__.
 
 Lambda expressions (Java)
-^^^^^^^^^^^^^^^^^^^^^^^^^
-
-While method references work well for passing a subroutine that has
-already been written, often it is inconvenient/wasteful to write a
-subroutine solely for the purpose of sending as a method reference, if
-that subroutine will never be used elsewhere. To avoid this, Java also
-supports a feature called “lambda expressions.” A lambda expression is
-an inline method definition - it allows a subroutine to be defined
-*inside of a parameter list*. For specifics on how to write lambda
-expressions, see `this
-tutorial <http://tutorials.jenkov.com/java/lambda-expressions.html>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+While method references work well for passing a subroutine that has already been written, often it is inconvenient/wasteful to write a subroutine solely for the purpose of sending as a method reference, if that subroutine will never be used elsewhere. To avoid this, Java also supports a feature called “lambda expressions.” A lambda expression is an inline method definition - it allows a subroutine to be defined *inside of a parameter list*. For specifics on how to write Java lambda expressions, see `this tutorial <http://tutorials.jenkov.com/java/lambda-expressions.html>`__
```

Comment:
This is a high risk resource because it is maintained by a single person. Is there a similar tutorial from Oracle or other organization?